packages/client/components/ActivityLibrary/ActivityDetailsCategoryBadge.tsx

```diff
@@ -0,0 +1,95 @@
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu'
+import graphql from 'babel-plugin-relay/macro'
+import clsx from 'clsx'
+import React from 'react'
+import {useFragment, useMutation} from 'react-relay'
+import {ActivityDetailsCategoryBadge_template$key} from '../../__generated__/ActivityDetailsCategoryBadge_template.graphql'
+import UpdateTemplateCategoryMutation, {
+  UpdateTemplateCategoryMutation as TUpdateTemplateCategoryMutation
+} from '../../__generated__/UpdateTemplateCategoryMutation.graphql'
+import useAtmosphere from '../../hooks/useAtmosphere'
+import PlainButton from '../PlainButton/PlainButton'
+import ActivityDetailsBadge from './ActivityDetailsBadge'
+import {CATEGORY_ID_TO_NAME, CATEGORY_THEMES, CategoryID} from './Categories'
+
+interface Props {
+  isEditing: boolean
+  templateRef: ActivityDetailsCategoryBadge_template$key
+}
+
+const ActivityDetailsCategoryBadge = (props: Props) => {
+  const {isEditing, templateRef} = props
+  const template = useFragment(
+    graphql`
+      fragment ActivityDetailsCategoryBadge_template on MeetingTemplate {
+        id
+        category
+      }
+    `,
+    templateRef
+  )
+  const {id: templateId} = template
+  const category = template.category as CategoryID
+  const [commit] = useMutation<TUpdateTemplateCategoryMutation>(UpdateTemplateCategoryMutation)
```

Comment:
thoughts on using `useMutation`? It has an `isInFlight` value that would allow us to stop using `useMutationProps`. I kinda dig it, but happy to switch back to old school, too.

Reply:
I kinda like it, too, but I'm hesitant to introduce it without a solid understanding of how it would fit in with our existing mutations, especially since this pattern would move the updater, onComplete, etc. methods into the component that calls them.

Let's use the existing pattern for now and look more deeply at this pattern separately.

Reply:
> especially since this pattern would move the updater, onComplete, etc. methods into the component that calls them.

This is true for `optimisticUpdater, updater`. We could move those back to the mutation file though since they don't have a scope.

`onError, onCompleted, variables` were already defined in the component, so there's no change for those, unless i'm misunderstanding what you meant?